### DISCUSSION TITLE
Can Virtual/Computed fields be returned to client?

### DISCUSSION THREAD

**Discussion by zgajo** [prisma/prisma#17859] [Q&A]
With the release of version 4.7, this feature has been added. 
I'm curious about the expected behavior for this. 
Can those fields be returned to the client, for example, Graphiql?

If I have model and computed field: 
```
prisma.schema
model BarberWorkTimeOverride {
  id       Int      @id @default(autoincrement())
  date     DateTime @db.Date()
  start    DateTime @db.Time()
  end      DateTime @db.Time()
  freeDay  Boolean  @default(false)
  barber   Barber   @relation(fields: [barberId], references: [id])
  barberId Int
}

prisma init
const prisma = new PrismaClient().$extends({
  result: {
    barberWorkTimeOverride: {
      end: {
        compute(barberWorkTimeOverride) {
          const date = moment(barberWorkTimeOverride.date).format('YYYY-MM-DD')
          const end = moment(barberWorkTimeOverride.end).format('HH:mm:ss')
          const combinedTime = moment(`${date}T${end}`).toISOString()
          return new Date(combinedTime)
        },
        needs: { date: true, end: true },
      },
      start: {
        compute(barberWorkTimeOverride) {
          const date = moment(barberWorkTimeOverride.date).format('YYYY-MM-DD')
          const start = moment(barberWorkTimeOverride.start).format('HH:mm:ss')
          const combinedTime = moment(`${date}T${start}`).toISOString()
          return new Date(combinedTime)
        },
        needs: { date: true, start: true },
      },
    },
  },
})
```

So accessing the `start` field from the backend will work as expected, it will return the computed value.
But, on graphiql it will show the value from db

```
const barberWorkTimeOverrides = await prisma.barberWorkTimeOverrides.findMany()

console.log(barberWorkTimeOverrides[0].start) // returns computed

```

```
const barberWorkTimeOverrides = await prisma.barberWorkTimeOverrides.findMany()
return barberWorkTimeOverrides // graphiql shows value from db

```

**Comment by nurul3101** [comment]
Hello @zgajo 👋 

Can you try changing the name of the computed field? You have named the computed fields as `end` and `start` which are existing fields on the BarberWorkTimeOverride model. Can you rename them to something else and check? You should be able to get the computed field in the output which you can send to the client.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
